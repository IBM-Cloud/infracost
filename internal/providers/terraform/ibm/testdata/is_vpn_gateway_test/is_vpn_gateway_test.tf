
terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "~> 1.40.0"
    }
  }
}

provider "ibm" {
    region = "us-south"
    generation = 2
}

resource "ibm_is_vpc" "testVpc" {
  name = "test-vpc"
}

resource "ibm_is_subnet" "testVpcSubnet" {
  name            = "example-subnet"
  vpc             = ibm_is_vpc.testVpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
}

resource "ibm_is_vpn_gateway" "testVpcVpn" {
  name   = "example-vpn-gateway"
  subnet = ibm_is_subnet.testVpcSubnet.id
  mode   = "route"
}