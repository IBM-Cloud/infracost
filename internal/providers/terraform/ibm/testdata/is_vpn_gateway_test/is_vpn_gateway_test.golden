
 Name                                     Monthly Qty  Unit      Monthly Cost 
                                                                              
 ibm_is_vpc.testVpc                                                           
 ├─ VPC instance                                    1  Instance         $0.00 
 └─ VPC egress us-south (first 9995 GB)             -  GB                   - 
 └─ VPC egress us-south (next 40000 GB)             -  GB                   - 
 └─ VPC egress us-south (next 100000 GB)            -  GB                   - 
 └─ VPC egress us-south (over 149995 GB)            -  GB                   - 
                                                                              
 ibm_is_vpn_gateway.testVpcVpn                                                
 ├─ VPN connection hours us-south                 730  Hours           $29.20 
 └─ VPN instance hours us-south                   730  Hours           $32.85 
                                                                              
 OVERALL TOTAL                                                         $62.05 
──────────────────────────────────
3 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 1 was free:
  ∙ 1 x ibm_is_subnet